### ISSUE TITLE
Update @babel/runtime to match material-ui project

### ISSUE THREAD

**Issue by ajorkowski** [oliviertassinari/react-event-listener#90]
Recently material-ui was updated to use babel/runtime@7.0.0-rc.1, however this project is pinned to babel/runtime@7.0.0-beta.42.

The main issue is that these two versions of babel/runtime are incompatible - so doing a `yarn install --flat` creates a breaking build :(

Here are all the errors that I am seeing when I try to force babel/runtime@7.0.0-rc.1 ... It looks like a  couple of things were removed...

```
ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/classCallCheck' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 7:38-94
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index

ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/createClass' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 8:35-88
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index

ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/inherits' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 10:32-82
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index

ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/objectSpread' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 13:36-90
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index

ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/objectWithoutProperties' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 12:47-112
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index

ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/possibleConstructorReturn' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 9:49-116
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index

ERROR in ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/builtin/typeof' in 'C:\projects\kalix.ui\client_modules\node_modules\react-event-listener\dist'
 @ ../client_modules/node_modules/react-event-listener/dist/react-event-listener.cjs.js 11:30-78
 @ ../client_modules/node_modules/@material-ui/core/Tabs/Tabs.js
 @ ../client_modules/node_modules/@material-ui/core/Tabs/index.js
 @ ../client_modules/node_modules/@material-ui/core/index.es.js
 @ ./app.js
 @ ./index.js
 @ multi ../node_modules/webpack-dev-server/client?http://localhost:3000 ../node_modules/webpack/hot/dev-server.js babel-polyfill index
```

**Comment by Fandekasp** [comment]
Similar error. When trying to upgrade to material-ui v1, I get errors like:
```
ERROR in ./node_modules/@material-ui/core/ButtonBase/ButtonBase.js
Module not found: Error: Can't resolve '@babel/runtime/helpers/getPrototypeOf' in 'project/node_modules/@material-ui/core/ButtonBase'
 @ ./node_modules/@material-ui/core/ButtonBase/ButtonBase.js 22:46-94
 @ ./node_modules/@material-ui/core/ButtonBase/index.js
 @ ./node_modules/@material-ui/core/Button/Button.js
 @ ./node_modules/@material-ui/core/Button/index.js
 @ ./src/components/pages/AccountDelete/index.js
 @ ./src/components/Routes.js
 @ ./src/components/Root.js
 @ ./src/app.js
 @ multi babel-regenerator-runtime ./src/app.js
```
I noted that while we're using Babel v6 for our project, material-ui got us to install babel v7 for its dev dependencies, but I couldn't see @babel/runtime in their packages.json requirements, so I tried to install it, then got errors such as the following:
```
ERROR in ./node_modules/@babel/runtime/helpers/builtin/classCallCheck.js       
Module build failed: Error: ENOENT: no such file or directory, open 'project/node_modules/@babel/runtime/helpers/builtin/classCallCheck.js'       
 @ ./node_modules/react-event-listener/dist/react-event-listener.cjs.js 7:38-94                                                                                                       
 @ ./node_modules/material-ui/Dialog/Dialog.js        
 @ ./node_modules/material-ui/Dialog/index.js      
 @ ./src/components/pages/AccountDelete/index.js
 @ ./src/components/Routes.js
 @ ./src/components/Root.js
 @ ./src/app.js
 @ multi babel-regenerator-runtime ./src/app.js

```

**Comment by oliviertassinari** [comment]
@ajorkowski I agree, we should be upgrading `@babel/runtime@7.0.0-rc.1`, however, I doubt you will ever be able to have matching versions of `babel/runtime` in your package.json. The version of this dependency should match the  `@babel/core`  version, something libraries authors use internally in development.

**Comment by farfromrefug** [comment]
@oliviertassinari same issue here trying to use yarn --flat.
About you comment. If it is a dev dev issue, we should not see it in production? And I always install in production mode, and I get the error ...

**Comment by oliviertassinari** [comment]
@farfromrefug As far as I understand the problem.
`@babel/runtime` (dependency) and `@babel/plugin-transform-runtime` (dev dependency) needs to have the same version to work. But there is no way you can enforce the community to have the same `@babel/plugin-transform-runtime` version installed. The usage of `@babel/plugin-transform-runtime` is common among library authors, as far as I understand the situation, you can forget about `yarn --flat` as long as some of your dependencies use it.

**Comment by farfromrefug** [comment]
@oliviertassinari I understand, bad to have version dependency between dev and prod. But I get the issue.
Indeed this is an issue for yarn --flat.  This also might get solved 7.0.0 release where things might stop to "move" so much.

**Comment by oliviertassinari** [comment]
Right, I will update the dependency here. Hopefully, things will stop moving.
